src/include/duckdb/common/single_thread_ptr.hpp

```diff
@@ -123,10 +125,11 @@ class single_thread_ptr {
 template <class T>
 struct _object_and_block {
 	T object;
-	uint32_t pn = 1;
+	uint32_t *pn;
```

Comment:
The point of the object_and_block is that the object and the reference counter are both in the same memory region, no? Turning this into a pointer means that this is no longer the case, which adds an additional layer of indirection for accessing the reference count. This seems to defeat the point of the _object_and_block class.

Could you have a look at fixing this without changing this into a pointer?

Reply:
Sure.